### DISCUSSION TITLE
Website hangs after full (8 months) XLS export

### DISCUSSION THREAD

**Discussion by mauricebar** [kimai/kimai#2438] [General]
**Describe the bug**
Weekly I need to pull a complete (8 month database) export in Excel. This is done in the Export Tab and every time it needs a lot of time and 99% of the times it does not complete it. Just a few times after many tries it delivers the file, which is actually pretty small, aprox. 800 Kb. I checked the server resources and it is quite free, CPU usage 27% and RAM used ~ 2 GB from 4GB (VM in Hyper-V)

**To Reproduce**
1. Go to 'Export'
2. Choose start date on 'Aug 01 2020' and end date 'today'
3. Click on 'Excel'
4. The wesite will go to white page or "500 - Fehler", many time I need to restart the server. 

**Logfile**
```
In attachment
```

**Additional context**
 - Kimai version:  1.13 stable (Ayumi) 
 - PHP version: 7.4.3
[log Kimai.txt](https://github.com/kevinpapst/kimai2/files/6141494/log.Kimai.txt)

 - Device: Linux sv02-ubuntu 5.4.0-66-generic #74-Ubuntu SMP Wed Jan 27 22:54:38 UTC 2021 x86_64
 - Browser Firefox 86.0.1

**Comment by kevinpapst** [comment]
> many time I need to restart the server

With all due respect: this is simply wrong. You never have to restart a server running PHP and MySQL when something crashes. Worst that can happen would be a restart of the PHP process (in case of an endless loop and a bad PHP configuration).

PHP (in normal web environments) can never use more than one CPU. So 27% of a 4-core machine is 100% in that one process.

Your log shows OutOfMemory errors. That should be a clear indication of what is wrong.
You need to increase the max memory limit of PHP.

And the main problem is probably your workflow which needs some adjustments. 
It should not be necessary to export the same data over and over again.
